TESTS/netsocket/dns/asynchronous_dns_cache.cpp

```diff
@@ -52,7 +52,7 @@ void ASYNCHRONOUS_DNS_CACHE()
 
         semaphore.acquire();
 
-        TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, data.result);
+        TEST_ASSERT_EQUAL(1, data.result);
```

Comment:
I suggest we either add a comment explaining this means "1 result was returned", or we give this "1" a name, such as:
```
constexpr int RESULTS_NUM 1
```